File: go-sdk/pkg/models/operations/getrooms.go
package operations

import (
	"openapi/pkg/models/shared"
)

type GetRoomsQueryParams struct {
	GroupIds      *string  `queryParam:"style=form,explode=true,name=groupIds"`
	Limit         *float64 `queryParam:"style=form,explode=true,name=limit"`
	MetadataKEY   *string  `queryParam:"style=form,explode=true,name=metadata.KEY"`
	StartingAfter *string  `queryParam:"style=form,explode=true,name=startingAfter"`
	UserID        *string  `queryParam:"style=form,explode=true,name=userId"`
}

type GetRoomsRequest struct {
	QueryParams GetRoomsQueryParams
}

type GetRoomsResponse struct {
	ContentType string
	Error       *shared.Error
	GetRooms    *shared.GetRooms
	StatusCode  int64
}

File: typescript-sdk/src/sdk/models/operations/getrooms.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetRoomsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=groupIds" })
  groupIds?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=metadata.KEY" })
  metadataKEY?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=startingAfter" })
  startingAfter?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=userId" })
  userId?: string;
}


export class GetRoomsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetRoomsQueryParams;
}


export class GetRoomsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  error?: shared.ErrorT;

  @SpeakeasyMetadata()
  getRooms?: shared.GetRooms;

  @SpeakeasyMetadata()
  statusCode: number;
}
